.github/workflows/regression-tests.yml

```diff
@@ -0,0 +1,42 @@
+name: Regression tests
+
+on:
+  push:
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  regression-tests-linux-mac:
+    name: Run on ${{ matrix.os }} using ${{ matrix.compiler }}
+    runs-on: ${{ matrix.os }}
+    env:
+      CXX: ${{ matrix.compiler }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        compiler: [g++-10, g++-13]
```

Comment:
Clang-12 can in principle be installed and used on the runner. However, I had issues analogous to [those](https://github.com/actions/runner-images/issues/8659#issuecomment-1783786560) when I tried to use it, hence it is not included yet. On my Ubuntu I have no issues using that version.

Reply:
Yeah, you often need to use the libc++ shipped with Clang, or an older libstdc++ version.
That's specially true if you live at HEAD.